harness/tests/src/main/kotlin/godot/tests/Invocation.kt

```diff
@@ -6,59 +6,112 @@ import godot.Spatial
+    @RegisterProperty var y = 0.0
+    @RegisterProperty var z = 0.0f
+    @RegisterProperty var customName = "Idonthaveanyidea"
 
-	var invocation = OtherScript()
+    var invocation = OtherScript()
 
-	@RegisterFunction fun intValue(value: Int) = value
-	@RegisterFunction fun longValue(value: Long) = value
-	@RegisterFunction fun floatValue(value: Float) = value
-	@RegisterFunction fun doubleValue(value: Double) = value
-	@RegisterFunction fun booleanValue(value: Boolean) = value
-	@RegisterFunction fun stringValue(value: String) = value
+    @RegisterSignal
+    val signalNoParam by signal()
 
-	@RegisterFunction
-	fun intAddition(a: Int, b: Int) = a + b
+    @RegisterSignal
+    val signalOneParam by signal<Boolean>("refresh")
 
-	@RegisterFunction
-	fun _enterTree() {
-		println("Enter tree !")
-		println("Instance id: ${getInstanceId()}")
-	}
+    @RegisterSignal
+    val signalTwoParam by signal<String, Invocation>("str", "inv")
 
     @RegisterFunction
-    fun _ready() {
-        val formerName = name
-        println("Name is: $name")
-        name = "TestName"
-        println("Name is: $name")
-        name = formerName
-        val test = DateTime.now() //external dependency to test dependency inclusion in dummyCompilation
-    }
+    fun intValue(value: Int) = value
 
-    override fun _onInit() {
-        println("Hello Invocation!")
-    }
+    @RegisterFunction
+    fun longValue(value: Long) = value
 
-    override fun _onDestroy() {
-        invocation.free()
-    }
+    @RegisterFunction
+    fun floatValue(value: Float) = value
 
-    //Type cast checks
     @RegisterFunction
-    fun parentIsSpatial() = getParent() is Spatial
+    fun doubleValue(value: Double) = value
+
     @RegisterFunction
-    fun isObjectSpatial(obj: Object) = obj is Spatial
+    fun booleanValue(value: Boolean) = value
+
+    @RegisterFunction
+    fun stringValue(value: String) = value
+
+    @RegisterFunction
+    fun intAddition(a: Int, b: Int) = a + b
+
+    @RegisterFunction
+    fun _enterTree() {
+        println("Enter tree !")
+        println("Instance id: ${getInstanceId()}")
+    }
+
+    @RegisterFunction
+    fun _ready() {
+		val formerName = name
+		println("Name is: $name")
+		name = "TestName"
+		println("Name is: $name")
+		name = formerName
+		val test = DateTime.now() //external dependency to test dependency inclusion in dummyCompilation
+
+		signalNoParam.connect(invocation, invocation::hookNoParam)
+		signalOneParam.connect(invocation, invocation::hookOneParam)
+		signalTwoParam.connect(invocation, invocation::hookTwoParam)
+		signalNoParam.emit()
+		signalOneParam.emit(false)
+		signalTwoParam.emit("My Awesome param !", this)
+	}
+
+	override fun _onInit() {
+		println("Hello Invocation!")
+	}
+
+	override fun _onDestroy() {
+		invocation.free()
+	}
+
+	//Type cast checks
+	@RegisterFunction
+	fun parentIsSpatial() = getParent() is Spatial
+	@RegisterFunction
+	fun isObjectSpatial(obj: Object) = obj is Spatial
```

Comment:
tabs :P

Reply:
mmmh weird, still got spaces in intellij.